src/main/java/io/github/thunkware/vt/bridge/ThreadNameCallable.java

```diff
@@ -0,0 +1,40 @@
+package io.github.thunkware.vt.bridge;
+
+import java.util.concurrent.Callable;
+
+/**
+ * Class to implement a NamedCallable that sets
+ * the thread name, executes the Callable, then
+ * resets the thread name
+ */
+public class ThreadNameCallable<V> implements Callable<V> {
```

Comment:
What do you think about making this as package-private and then exposing public factory methods in [ThreadCustomizer](https://github.com/thunkware/virtual-threads-bridge/blob/a539480ed52da3c4a89370cb0b8bb45179f464a9/src/main/java/io/github/thunkware/vt/bridge/ThreadCustomizer.java#L9C18-L9C35)? I don't have strong feelings for that but I am curious

something like:

```java
/**
 * Callback interface for customizing u̶n̶s̶t̶a̶r̶t̶e̶d̶ threads
 */
@FunctionalInterface
public interface ThreadCustomizer {

    void customize(Thread u̶n̶s̶t̶a̶r̶t̶e̶d̶T̶h̶r̶e̶a̶d̶ thread);

    default ThreadFactory asThreadFactory(ThreadFactory delegate) { ... }
    static ThreadCustomizer withNamePrefix(String prefix) { ... }

    /**
     * Create a Callable wrapper that sets thread names, executes given callable, and 
     * then resets the thread name
     */
    static Callable<T> withName(String name, Callable<T> delegate) {
          return ...
    }
}
```

Reply:
I could see a scenario where someone may want to set the thread name and not reset it. Just based on method names, that's what I would expect to happen if using these methods...

```java
static Runnable withName(String name, Runnable delegate) {
          return ...
    }
```

```java
static Callable<T> withName(String name, Callable<T> delegate) {
          return ...
    }
```

The challenge is how to name the method to indicate it's a temporary change/resets the thread name after execution.

Thoughts on something like...

```java
static Runnable withEphemeralName(String name, Runnable delegate) {
          return ...
    }
```

```java
static Callable<T> withEphemeralName(String name, Callable<T> delegate) {
          return ...
    }
```

Reply:
Also, the code could use anonymous inner classes for implementation...

```java
    /**
     * Create a {@link Runnable} that sets the thread name used during execution,
     * restoring the original thread name after execution
     *
     * @param name ephemeral thread name
     * @param task task
     * @return {@link Runnable}
     */
    static Runnable withEphemeralName(String name, Runnable task) {
        if (name == null) {
            throw new IllegalArgumentException("name is null");
        }

        if (task == null) {
            throw new IllegalArgumentException("task is null");
        }

        return () -> {
            Thread currentThread = Thread.currentThread();
            String originalThreadName = currentThread.getName();

            try {
                currentThread.setName(name);
                task.run();
            } finally {
                currentThread.setName(originalThreadName);
            }
        };
    }

    /**
     * Create a {@link Callable} that sets the thread name used during execution,
     * restoring the original thread name after execution
     *
     * @param name ephemeral thread name
     * @param callable callable
     * @return {@link Callable}
     */
    static <T>Callable<T> withEphemeralName(String name, Callable<T> callable) {
        if (name == null) {
            throw new IllegalArgumentException("name is null");
        }

        if (callable == null) {
            throw new IllegalArgumentException("callable is null");
        }

        return () -> {
            Thread currentThread = Thread.currentThread();
            String originalThreadName = currentThread.getName();

            try {
                currentThread.setName(name);
                return callable.call();
            } finally {
                currentThread.setName(originalThreadName);
            }
        };
    }
```

Reply:
Fair. It's a bit of square peg and round hole